### ISSUE TITLE
Finer-grained control over output of Attachments

### ISSUE THREAD

**Issue by goofballLogic** [cucumber/cucumber-js#1160]
This seems to be the same as a recent closed/abandoned issue: #1148 

Per discussion https://stackoverflow.com/questions/53613111/cucumber-js-prevent-attachments-swamping-output/53618285?noredirect=1#comment94135284_53618285 this is a feature request to control whether or not attachments get automatically output from the default reporters (like Progress).

Per the documentation (https://github.com/cucumber/cucumber-js/blob/master/docs/support_files/attachments.md):

> Attachments are also printed by the progress, progress-bar and summary formatters. They appear right after the step and only text/plain content is visible. It can be used to debug scenarios, especially in parallel mode.

However, sometimes I attach text/plain content which I don't want cluttering up the output, especially on a minimalist fomatter like progress.

A practical example from my current project is the HTML of a web page which I am testing via cucmberjs -> Selenium. This only gets attached in the event of a failing scenario, and is only used if a test run fails in the context of a continuous delivery pipeline. When running the tests locally, developers don't want that massive attachment swamping all other screen output. (text/plain content type is needed for compatibility with downstream reporters we use later in our processes).

I note that the decision to _only_ emit text/plain attachments seems somewhat arbitrary (albeit a good compromise). It would be nice if I had some control over what attachments get output. I'm not sure if this is a feature request for cucumberjs or whether it's for one/more formatters.

**Comment by Izhaki** [comment]
Possible duplicate of #1136 ?

**Comment by goofballLogic** [comment]
Agreed @Izhaki will add a note there and close this one.
